src/credentials/tests/TestChipCert.cpp

```diff
@@ -1113,70 +1113,68 @@ static void TestChipCert_ExtractCATsFromOpCert(nlTestSuite * inSuite, void * inC
     struct TestCase
     {
         uint8_t Cert;
-        uint32_t ExpectedCAT[kMaxSubjectCATAttributeCount];
+        CATValues ExpectedCATs;
     };
 
     // clang-format off
     static constexpr TestCase sTestCases[] = {
         // Cert                  CATs
-        // =============================================================
-        {  TestCert::kNode01_01, {          0,          0,          0 } },
-        {  TestCert::kNode01_02, {          0,          0,          0 } },
-        {  TestCert::kNode02_01, {          0,          0,          0 } },
-        {  TestCert::kNode02_02, {          0,          0,          0 } },
-        {  TestCert::kNode02_03, { 0xABCD0001,          0,          0 } },
-        {  TestCert::kNode02_04, { 0xABCE1002, 0xABCD0003,          0 } },
-        {  TestCert::kNode02_05, { 0xABCD0010, 0xABCE1008,          0 } },
-        {  TestCert::kNode02_06, {          0,          0,          0 } },
-        {  TestCert::kNode02_07, {          0,          0,          0 } },
-        {  TestCert::kNode02_08, { 0xABCF00A0, 0xABCD0020, 0xABCE0100 } },
+        // ============================================================================
+        {  TestCert::kNode01_01, { { kUndefinedCAT, kUndefinedCAT, kUndefinedCAT } } },
+        {  TestCert::kNode01_02, { { kUndefinedCAT, kUndefinedCAT, kUndefinedCAT } } },
+        {  TestCert::kNode02_01, { { kUndefinedCAT, kUndefinedCAT, kUndefinedCAT } } },
+        {  TestCert::kNode02_02, { { kUndefinedCAT, kUndefinedCAT, kUndefinedCAT } } },
+        {  TestCert::kNode02_03, { {    0xABCD0001, kUndefinedCAT, kUndefinedCAT } } },
+        {  TestCert::kNode02_04, { {    0xABCE1002,    0xABCD0003, kUndefinedCAT } } },
+        {  TestCert::kNode02_05, { {    0xABCD0010,    0xABCE1008, kUndefinedCAT } } },
+        {  TestCert::kNode02_06, { { kUndefinedCAT, kUndefinedCAT, kUndefinedCAT } } },
+        {  TestCert::kNode02_07, { { kUndefinedCAT, kUndefinedCAT, kUndefinedCAT } } },
+        {  TestCert::kNode02_08, { {    0xABCF00A0,    0xABCD0020,    0xABCE0100 } } },
```

Comment:
Is it valid for the same CAT to appear with multiple versions? I don't see that here, so it's not a problem here, but seeing this list where I had to look closely caused me the question

Reply:
Good question. 
On one side, it is not reasonable for CAs to issue DAC with multiple CATs with same identifier but different versions. On the other, I don't see any functional issue with that case as the Access Control code will be looking for the correct version number until it finds one. I don't think this case requires any special treatment.